narwhals/_duckdb/dataframe.py

```diff
@@ -1,148 +1,172 @@
+            )
+        )
 
-    def __getattr__(self, attr: str) -> Any:
-        if attr == "schema":
-            return {
-                column_name: native_to_narwhals_dtype(str(duckdb_dtype), self._version)
-                for column_name, duckdb_dtype in zip(
-                    self._native_frame.columns, self._native_frame.types
-                )
-            }
-        elif attr == "columns":
-            return self._native_frame.columns
-
-        msg = (  # pragma: no cover
-            f"Attribute {attr} is not supported for metadata-only dataframes.\n\n"
-            "If you would like to see this kind of object better supported in "
-            "Narwhals, please open a feature request "
-            "at https://github.com/narwhals-dev/narwhals/issues."
+    def drop(self: Self, columns: list[str], strict: bool) -> Self:  # noqa: FBT001
+        columns_to_drop = parse_columns_to_drop(
+            compliant_frame=self, columns=columns, strict=strict
         )
-        raise NotImplementedError(msg)  # pragma: no cover
+        selection = (col for col in self.columns if col not in columns_to_drop)
+        return self._from_native_frame(self._native_frame.select(*selection))
+
+    def lazy(self) -> Self:
+        return self
+
+    def with_columns(
+        self: Self,
+        *exprs: Any,
+        **named_exprs: Any,
+    ) -> Self:
+        from duckdb import ColumnExpression
+
+        new_columns_map = parse_exprs_and_named_exprs(self, *exprs, **named_exprs)
+        result = []
+        for col in self._native_frame.columns:
+            if col in new_columns_map:
+                result.append(new_columns_map.pop(col).alias(col))
+            else:
+                result.append(ColumnExpression(col))
+        for col, value in new_columns_map.items():
+            result.append(value.alias(col))
+        return self._from_native_frame(self._native_frame.select(*result))
+
+    def filter(self, *predicates: DuckDBExpr, **constraints: Any) -> Self:
+        plx = self.__narwhals_namespace__()
+        expr = plx.all_horizontal(
+            *chain(predicates, (plx.col(name) == v for name, v in constraints.items()))
+        )
+        # `[0]` is safe as all_horizontal's expression only returns a single column
+        mask = expr._call(self)[0]
+        return self._from_native_frame(self._native_frame.filter(mask))
+
+    @property
+    def schema(self) -> dict[str, DType]:
+        return {
+            column_name: native_to_narwhals_dtype(str(duckdb_dtype), self._version)
+            for column_name, duckdb_dtype in zip(
+                self._native_frame.columns, self._native_frame.types
+            )
+        }
+
+    @property
+    def columns(self) -> list[str]:
+        return self._native_frame.columns  # type: ignore[no-any-return]
 
     def to_pandas(self: Self) -> pd.DataFrame:
+        # only is version if v1, keep around for backcompat
```

Comment:
TODO: implement version check? Same for `to_arrow`?

```suggestion
        # only if version is v1, keep around for backcompat
```

Reply:
`to_pandas` wouldn't be available on `nw.LazyFrame` anyway so this wouldn't be reachable for non-v1